lib/api/v1/controllers/authentication.go

```diff
@@ -22,25 +25,35 @@ func (c *AuthenticationController) Login(ctx *gin.Context) {
 
 	err := ctx.ShouldBindWith(authenticationForm, binding.Form)
 	if err != nil {
-		ResponseWithError(ctx, http.StatusBadRequest, err)
+		ResponseWithError(ctx, http.StatusBadRequest, err, constants.ERROR_CODE_MALFORM_REQUEST)
 		return
 	}
 
 	_, err = authenticationForm.Validate()
 	if err != nil {
-		ResponseWithError(ctx, http.StatusBadRequest, err)
+		ResponseWithError(ctx, http.StatusBadRequest, err, constants.ERROR_CODE_INVALID_PARAM)
 		return
 	}
 
 	err = authenticationForm.ValidateUser()
 	if err != nil {
-		ResponseWithError(ctx, http.StatusUnauthorized, err)
+		ResponseWithError(ctx, http.StatusUnauthorized, err, constants.ERROR_CODE_INVALID_CREDENTIALS)
 		return
 	}
 
-	err = oauth.HandleTokenRequest(ctx)
+	tokenData, err := oauth.HandleTokenRequest(ctx)
 	if err != nil {
-		ResponseWithError(ctx, http.StatusUnauthorized, errors.New(constants.OAuthClientInvalid))
+		ResponseWithError(ctx, http.StatusUnauthorized, errors.New(constants.OAuthClientInvalid), constants.ERROR_CODE_INVALID_CREDENTIALS)
 		return
 	}
+
+	response := &serializers.AuthenticationResponse{
+		ID:           int64(rand.Uint64()),
+		AccessToken:  tokenData["access_token"].(string),
```

Comment:
Should we add a serializer model and use `unmarshal` to parse this object instead?

```
type TokenSerializer struct {
	AccessToken  string `json:"access_token"`
	ExpireIn     int64  `json:"expires_in"`
	RefreshToken string `json:"refresh_token"`
	TokenType    string `json:"token_type"`
}

tokenSerializer := serializers.TokenSerializer{}
err = json.Unmarshal(tokenData, &tokenSerializer)
```
Then you can use `tokenSerializer` to pass into the response 🤔